#VRML_SIM R2021b utf8
WorldInfo {
  info [
    "This demonstration shows a DJI Mavic 2 PRO flying over a rural area."
    "After the drone has taken off, you can control it with the keyboard of your computer."
  ]
  title "DJI Mavic 2 PRO demo"
  basicTimeStep 8
  defaultDamping Damping {
    linear 0.5
    angular 0.5
  }
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.2569256657081811 -0.9119103752066338 -0.3200135464178115 1.8776541464939822
  position -4.532880052839744 3.6883567955610057 -1.4102684122777709
  near 0.2
  followSmoothness 0.2
}
TexturedBackground {
  luminosity 3
}
TexturedBackgroundLight {
}
Floor {
  size 400 400
  tileSize 6 6
  appearance SandyGround {
  }
}
Mavic2Pro {
  hidden position_0_0 1.3373605273699564e-09
  hidden position_14_0 -3.112123891924057e-09
  hidden position_15_0 1.1851495415047495e-09
  translation -0.004785374911454365 0.0857087278743042 0.02000000383482182
  rotation -0.9987852429767983 -0.03484279907696976 0.03484275770171151 1.5720107045450524
  name "Mavic 2 PRO_0"
  controller "base_controller_2"
  bodySlot [
    Emitter {
      model "A"
      channel -1
    }
    Receiver {
      model "A"
    }
    Connector {
      translation -0.07 0 -0.15
      rotation 0 1 0 -3.1
      name "magnetic"
      model "a"
      type "active"
      autoLock TRUE
      distanceTolerance 0.3
      axisTolerance 1.5
      rotationTolerance 1.5
      snap FALSE
    }
    DistanceSensor {
      translation 0.08 0 0.01
      children [
        Shape {
          appearance DEF App_s PBRAppearance {
            baseColor 0 0 1
          }
          geometry DEF geo_sems Box {
            size 0.01 0.01 0.005
          }
        }
      ]
      name "front sensor"
      type "sonar"
      numberOfRays 5
    }
    DistanceSensor {
      translation -0.03 0 0.02
      rotation 0 1 0 -1.5707953071795862
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.01 0.01 0.005
          }
        }
      ]
      name "upper sensor"
      type "sonar"
      numberOfRays 5
    }
    DistanceSensor {
      translation -0.0386371 0 -0.0603528
      rotation 0 1 0 1.309
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "lower sensor"
      type "sonar"
      numberOfRays 5
    }
    DistanceSensor {
      translation -0.0200002 -0.04 -0.02
      rotation 0 0 -1 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.005
          }
        }
      ]
      name "right sensor"
      type "sonar"
      numberOfRays 5
    }
    DistanceSensor {
      translation -0.0200002 0.04 -0.02
      rotation 0.707108281185553 0.70710528118436 9.236343672883591e-09 -3.1415853071795863
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.005
          }
        }
      ]
      name "left sensor"
      type "sonar"
      numberOfRays 5
    }
  ]
}
DEF SUPERVISOR Robot {
  children [
    Emitter {
      model "A"
      channel -1
      byteSize 4096
    }
  ]
  name "supervisor"
  controller "supervisor_controller"
  supervisor TRUE
}
Road {
  translation -43.8 0.01 -9.2
  rotation 0 1 0 1.1781
  width 3
  numberOfLanes 1
  rightBorder FALSE
  leftBorder FALSE
  wayPoints [
    0 0 0
    0 0 100
    200 0 200
  ]
  splineSubdivision 8
  appearance Pavement {
    type "black stone"
  }
}
Windmill {
  translation -43.8656 0 19.8433
  rotation 0 1 0 -1.0472
}
Windmill {
  translation -44.258 0 27.3356
  rotation 0 1 0 -1.0472
  name "windmill(1)"
}
Windmill {
  translation -46.2448 0 -30.573
  rotation 0 1 0 -1.0472
  name "windmill(2)"
}
Windmill {
  translation -38.75 0 -23.4
  rotation 0 1 0 -1.5708
  name "windmill(3)"
}
DEF base_drone SquareManhole {
  translation 0 -0.03 0
  name "base_drone"
}
DEF base_drone SquareManhole {
  translation 10 -0.03 10
  name "base_drone(2)"
}
DEF base_drone SquareManhole {
  translation 0 -0.03 1
  name "base_drone(1)"
}
DEF base_1 Solid {
  translation 3.18 0.04019 2.17
  rotation 1.4057796113349329e-18 -0.9999999999999999 -1.2632965782802954e-18 1.5707953071795862
  children [
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "../textures/base1.png"
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Box {
        size 1 0.1 1
      }
    }
  ]
  contactMaterial "base"
  boundingObject Box {
    size 1 0.1 1
  }
  physics Physics {
  }
  linearVelocity 2.671466538073522e-18 -1.8427469958982662e-16 2.64069195238033e-19
  angularVelocity 3.924439319495078e-17 8.821390035672937e-19 -4.2414111460963625e-17
}
DEF base_2 Solid {
  translation 3.18 0.040189018999999986 -1.76
  rotation -6.800071135043973e-18 -0.9999999999999999 5.431471499640125e-18 1.5707953071795862
  children [
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "../textures/base2.png"
          ]
        }
        roughness 1
        metalness 0
      }
      geometry Box {
        size 1 0.1 1
      }
    }
  ]
  name "solid(1)"
  contactMaterial "base"
  boundingObject Box {
    size 1 0.1 1
  }
  physics Physics {
  }
  linearVelocity -7.368707901187463e-19 6.706369552522968e-17 -2.1341649849093593e-18
  angularVelocity -4.597812825602895e-17 6.570396789523675e-19 2.6557226351465318e-17
}
Pine {
  translation -14.0083 0 14.4828
  rotation 0 1 0 3.14159
}
Pine {
  translation -19.6195 0 24.3783
  rotation 0 1 0 2.87979
  name "pine tree(1)"
}
Pine {
  translation -22.0129 0 -6.20205
  name "pine tree(3)"
}
Pine {
  translation -10.6926 0 25.1339
  name "pine tree(4)"
}
Pine {
  translation -14.3441 0 -14.5621
  name "pine tree(5)"
}
Pine {
  translation -26.6328 0 7.1707
  name "pine tree(6)"
}
Pine {
  translation -9.37029 0 -14.0209
  rotation 0 1 0 -2.618
  name "pine tree(2)"
}
Forest {
  translation 0 -0.01 0
  shape [
    200 200
    200 0
    50 0
    50 50
    -50 50
    -50 -50
    50 -50
    50 0
    200 0
    200 -200
    -200 -200
    -200 200
  ]
  density 0.01
  type "cherry tree"
  randomSeed 1
  maxHeight 12
  minHeight 6
  maxRadius 6
  minRadius 3
}
Solid {
  translation 0.03 0.04 0.0099999
  rotation 1 0 0 4.6928e-06
  children [
    Charger {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0
            roughness 1
            metalness 0
            emissiveColor 0 0.40083199739456177 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.45
          }
        }
      ]
      name "charger 1"
      battery [
        125.51999999999946, 500, 10
      ]
      radius 0.45
    }
  ]
  name "Charger_station"
  recognitionColors [
    0 0 0
  ]
}
DEF order_box Solid {
  translation 3.18 0.19018803800000003 -1.76
  rotation -0.012347912206131326 0.8535040752382838 -0.5209398454867814 3.8095234463548096e-17
  children [
    Connector {
      translation 0 0.2 3.3671e-08
      rotation 1 0 0 -1.5707996938995747
      model "a"
      type "passive"
      autoLock TRUE
      distanceTolerance 0.7
      axisTolerance 1.5
    }
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "../textures/amazon-texture.jpg"
          ]
          repeatS FALSE
          repeatT FALSE
        }
        roughness 1
        metalness 0
      }
      geometry DEF BOX_GEOMETRY Box {
        size 0.3 0.2 0.3
      }
    }
  ]
  name "box"
  contactMaterial "box"
  boundingObject USE BOX_GEOMETRY
  physics Physics {
    mass 0.01
  }
  linearVelocity 1.0733346104350752e-18 -1.026797821064395e-15 5.743545882884969e-18
  angularVelocity 6.369713530176149e-17 2.0859551521602186e-17 3.184856765088074e-17
}
